File: src/main/java/no/fintlabs/integration/validation/constraints/ActiveConfigurationIsDefinedIfStateIsActive.java
package no.fintlabs.integration.validation.constraints;

import jakarta.validation.Constraint;
import jakarta.validation.Payload;
import java.lang.annotation.Retention;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Retention(RUNTIME)
@Constraint(validatedBy = ActiveConfigurationIsDefinedIfStateIsActiveValidator.class)
public @interface ActiveConfigurationIsDefinedIfStateIsActive {

    String message() default "is cannot be set to ACTIVE without an active configuration";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};

}

File: src/main/kotlin/no/novari/flyt/integration/validation/constraints/ActiveConfigurationIsDefinedIfStateIsActive.kt
package no.novari.flyt.integration.validation.constraints

import jakarta.validation.Constraint
import jakarta.validation.Payload
import kotlin.reflect.KClass

@Target(AnnotationTarget.FIELD, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.VALUE_PARAMETER)
@Retention(AnnotationRetention.RUNTIME)
@Constraint(validatedBy = [ActiveConfigurationIsDefinedIfStateIsActiveValidator::class])
annotation class ActiveConfigurationIsDefinedIfStateIsActive(
    val message: String = "is cannot be set to ACTIVE without an active configuration",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = [],
)
